core/src/main/java/org/apache/iceberg/UpdateRequirement.java

```diff
@@ -135,6 +147,15 @@ public void validate(TableMetadata base) {
             schemaId, base.currentSchemaId());
       }
     }
+
+    @Override
+    public void validate(ViewMetadata base) {
+      if (schemaId != base.currentSchemaId()) {
```

Comment:
This looks correct, but would we ever use it? You can't evolve the schema of a view because the schema must match what the version produces. That means the schema always changes with the view version (or is reused) and there isn't a need to use this validation.